exla/lib/exla/backend.ex

```diff
@@ -0,0 +1,268 @@
+  def backend_copy(%T{data: %B{buffer: buffer}} = tensor, backend, opts) do
+    backend.from_binary(tensor, EXLA.DeviceBuffer.read(buffer), opts)
+  end
+
+  @impl true
+  def backend_transfer(%T{data: %B{buffer: buffer}} = tensor, backend, opts) do
+    if backend == __MODULE__ and same_client_device?(buffer, opts) do
+      tensor
+    else
+      try do
+        backend_copy(tensor, backend, opts)
+      after
+        EXLA.DeviceBuffer.deallocate(buffer)
+      end
+    end
+  end
+
+  @impl true
+  def backend_deallocate(%T{data: %B{buffer: buffer}}) do
+    EXLA.DeviceBuffer.deallocate(buffer)
+  end
+
+  @impl true
+  def to_batched_list(out, tensor, opts) do
+    binary_tensor = backend_copy(tensor, Nx.BinaryBackend, [])
+    Nx.BinaryBackend.to_batched_list(out, binary_tensor, opts)
+  end
+
+  @impl true
+  def to_binary(%T{data: %B{buffer: buffer}, type: {_, size}}, limit) do
+    EXLA.DeviceBuffer.read(buffer, limit * div(size, 8))
+  end
+
+  @impl true
+  def inspect(%T{} = tensor, inspect_opts) do
+    result =
+      if platform(tensor) == :host do
+        binary = to_binary(tensor, Nx.size(tensor))
+        Nx.Backend.inspect(tensor, binary, inspect_opts)
+      else
+        "Tensors on the GPU cannot be inspected. Explicitly transfer the tensor by calling Nx.backend_transfer/1"
+      end
+
+    maybe_add_signature(result, tensor)
+  end
+
+  # TODO: Elixir v1.13 has a default_inspect_fun which
+  # we can use to customize this behaviour for tests.
+  if Application.compile_env(:exla, :add_backend_on_inspect, true) do
+    defp maybe_add_signature(result, %T{data: %B{buffer: buffer}}) do
+      %EXLA.DeviceBuffer{client_name: client_name, device_id: device_id, ref: ref} = buffer
+      '#Ref<' ++ rest = :erlang.ref_to_list(ref)
+      info = "EXLA.Backend<#{client_name}:#{device_id}, " <> List.to_string(rest)
+      Inspect.Algebra.concat([info, Inspect.Algebra.line(), result])
+    end
+  else
+    defp maybe_add_signature(result, _tensor) do
+      result
+    end
+  end
+
+  ## Helpers
+
+  defp default_client_name do
+    opts = Nx.Defn.default_options()
+
+    if opts[:compiler] == EXLA do
+      opts[:client] || :host
+    else
+      :host
+    end
+  end
+
+  defp client_and_device_id(opts) do
+    client = EXLA.Client.fetch!(opts[:client] || default_client_name())
+    device_id = opts[:device_id] || client.default_device_id
+    {client, device_id}
+  end
+
+  defp same_client_device?(buffer, opts) do
+    {client, device_id} = client_and_device_id(opts)
+    buffer.client_name == client.name and buffer.device_id == device_id
+  end
+
+  defp platform(%T{data: %B{buffer: buffer}}) do
+    client = EXLA.Client.fetch!(buffer.client_name)
+    client.platform
+  end
+
+  @impl true
+  def eye(%{shape: shape, type: type, names: names}, _) do
+    fun = fn ->
+      Nx.eye(shape, type: type, names: names, backend: Nx.Defn.Expr)
+    end
+
+    EXLA.jit(fun, [])
+  end
```

Comment:
This variation wouldn't work?

```suggestion
  def eye(out, tensor) do
    fun = fn -> Nx.Defn.Expr.eye(out, tensor) end
    EXLA.jit(fun, [])
  end
```

The same below...

Reply:
I don't see a reason why it wouldn't, and if so, we can just generate it as all the others. Also for `random_uniform` and `random_normal` I think we want to pass the `min`, `max`, `sigma`, `mu` through JIT arguments. @seanmor5 please let us know if we miss something :D